lib/pushy_client/protocol_handler.rb

```diff
@@ -483,8 +483,14 @@ def self.send_signed_message(socket, method, client_private_key, session_key, me
              when :hmac_sha256
                make_header_hmac(message, session_key)
              end
-      socket.send_string(auth, ZMQ::SNDMORE)
-      socket.send_string(message)
+      begin
+        timeout(10) do
+          socket.send_string(auth, ZMQ::SNDMORE)
+          socket.send_string(message)
+        end
+      rescue
```

Comment:
Please rescue just `Timeout::Error` here. Since we aren't printing the exception in the message below, it is possible that some other exception would be getting raise and it would be blindly rescued here with an incorrect message about what happened.